### PR TITLE
feat: geodata reader

### PR DESCRIPTION
**Pull request by raphaelvignes**
Add a new geojson reader

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable

### CODE REVIEW

File: peakina/helpers.py [hunk 1]
```diff
@@ -42,7 +43,7 @@ class TypeInfos(NamedTuple):
 
 
 # For files without MIME types, we make fake MIME types based on detected extension
-CUSTOM_MIMETYPES = {".parquet": "peakina/parquet"}
+CUSTOM_MIMETYPES = {".parquet": "peakina/parquet", ".geojson": "application/geo+json"}
```
Comment (fspot) [inline peakina/helpers.py]: are we okay with enforcing this extension for basemaps ? cf. my comment [here](https://toucantoco.atlassian.net/wiki/spaces/TTA/pages/3498344456/PAT+248+-+Handle+geographic+data+%28basemaps%29+as+usual+datasources?focusedCommentId=3503226890#comment-3503226890)

File: peakina/readers/geojson.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from functools import wraps
+from typing import Any, Optional
+
+import geopandas as gpd
+
+
+@wraps(gpd.read_file)
+def read_file(
```
Comment (fspot) [inline peakina/readers/geojson.py]: maybe rename this function so it's explicit it's for geographic data

File: peakina/helpers.py [hunk 3]
```diff
@@ -81,6 +86,7 @@ class TypeEnum(str, Enum):
     JSON = "json"
     PARQUET = "parquet"
     XML = "xml"
+    GEOJSON = "geojson"
```
Comment (lukapeschke) [inline peakina/helpers.py]: I don't think we should have `GEOJSON` here. I'd rather have something like `GEODATA`, and let it be a wildcard for everything that can be read by geopandas (even if we only support geojson in laputa for now)

File: peakina/helpers.py (hunk 1 above)
Comment (lukapeschke) [reply peakina/helpers.py]: Is `application/geo+json` really a valid mimetype ? If not, I'd rather have `peakina/geo` to make it explicit that this is a custom mimetype. And I agree with fred's comment, the extension for geo files should probably be free (if that's possible in peakina)

**Review by lukapeschke — COMMENTED** [review]
Code LGTM overall, but I have a few remarks

File: peakina/helpers.py (hunk 1 above)
Comment (raphaelvignes) [reply peakina/helpers.py]: > if that's possible in peakina 

MimeType guessing is not working it seems :/

File: peakina/readers/geojson.py (hunk 2 above)
Comment (raphaelvignes) [reply peakina/readers/geojson.py]: alright :+1:

File: peakina/helpers.py (hunk 3 above)
Comment (raphaelvignes) [reply peakina/helpers.py]: okay

**Review by jGundermann — APPROVED** [review]
LGTM

**Pull request merged**